var/spack/repos/builtin/packages/matlab/package.py

```diff
@@ -45,8 +45,19 @@ class Matlab(Package):
 
     version('R2016b', 'b0e0b688894282139fa787b5a86a5cf7')
 
-    variant('mode', default='interactive', description='Installation mode (interactive, silent, or automated)')
-    variant('key',  default='',            description='The file installation key to use')
+    variant(
+        'mode',
+        default='interactive',
+        values=('interactive', 'silent', 'automated'),
+        description='Installation mode (interactive, silent, or automated)'
+    )
+
+    variant(
+        'key',
+        default='',
+        values=lambda x: True,  # Anything goes as a key
```

Comment:
Why is this line needed?  I would imagine that `variant()` should imply this if the user doesn't specify `values=...`. 

OK I see... nothing means a boolean variant, we probably need to keep that.  Still, users shouldn't have to write `lambda`.  Let's at least wrap this up into a nicely-named symbol somewhere.